lib/chef/resource/macos_userdefaults.rb

```diff
@@ -73,54 +107,90 @@ def coerce_booleans(val)
       end
 
       load_current_value do |desired|
-        value = coerce_booleans(desired.value)
-        cmd = "defaults read '#{desired.domain}' "
-        cmd << "'#{desired.key}' " if desired.key
-        cmd << " | grep -qx '#{value}'"
-
-        vc = if desired.user.nil?
-               shell_out(cmd)
-             else
-               shell_out(cmd, user: desired.user)
-             end
-
-        is_set !vc.error?
+        coerced_value = coerce_booleans(desired.value)
+
+        state_cmd = ["/usr/bin/defaults", "read", desired.domain, desired.key]
```

Comment:
the fact that key is required is new and backwards incompat... do we need to keep back-compat for that until 17?

Reply:
This resource had some pretty fundamental design flaws that require breaking it to fix.  The previous behavior where it wasn't required seems to 100% broken. You would only not provide the key if you were setting the global property to true and in that case we used key values in the domain property (wtf) and didn't specify a domain. It doesn't look like that ever worked. I tried to repro the examples and just got errors so I think anyone who ever tried to skip that property just had chef blow up in their face.